src/Widgets/NavigationWidget/SubComponents/LanguageSwitch.tsx

```diff
@@ -44,11 +45,8 @@ export const LanguageSwitch = connect(function LanguageSwitch() {
         {versions.map(({ version, root }) => (
           <NavDropdown.Item
             key={root.id()}
-            href={
-              version
-                ? urlFor(version, { query: currentQuery() })
-                : urlFor(root)
-            }
+            as={LinkTag}
+            to={new Link({ obj: version || root, query: currentQuery() })}
```

Comment:
* [ ] Instead of using the workaround of `new Link` we should use the `to` and `params` props. I think this will even simplify `currentQuery`.